### ISSUE TITLE
Summary separators rendered with incorrect indentation

### ISSUE THREAD

**Issue by OmarTawfik** [rust-lang/mdBook#1730]
Thanks for the awesome tool.
One issue I noticed, considering the following `SUMMARY.md` [separators feature](https://rust-lang.github.io/mdBook/format/summary.html):

Separators are using the indentation of the block before it, not after it, which is incorrect in this case:

```markdown
# Summary

- [Level One]()
  - [Example](./chapter_1.md)
 
---

- [Level Two]()
  - [Example](./chapter_1.md)
```

Which is rendered like this:

![image](https://user-images.githubusercontent.com/15987992/150076771-f6bd0208-9659-4daf-89b6-aa23b9f12b51.png)

I suggest calculating the indentation from the entry after it, not before it, in order to fix this behavior.

**Comment by OmarTawfik** [comment]
@ehuss happy to take a stab at this, but I'm not sure where to start. Any pointers would be appreciated.

**Comment by ehuss** [comment]
I suspect it would be around [here](https://github.com/rust-lang/mdBook/blob/0ec4b692f49b11d961e740e1ed14c4507d48621d/src/renderer/html_handlebars/helpers/toc.rs#L63-L67), where it uses an `<li>` for the spacer.  Perhaps it could use a different tag, like `<hr>`? However, this would need careful consideration of how it will change existing users, and whether or not it will affect their expected rendering.

**Comment by kg4zow** [comment]
I'm not sure if "me too" comments are appreciated, but I've noticed the same thing. Nobody *reading* my docs has mentioned it though, so I just figured it was just my "perfectionism" again (I tend to wrestle with the whole "perfect is the enemy of good" thing quite a bit).

FWIW I would suggest having the indendation of the separator bar in the HTML, somehow match the indentation of the first `-` in the `SUMMARY.md` text, so if the `---` line has four spaces before the first `-`, the separator bar would be indented at the same "level" as other items in the file which have four spaces before them.

Either that, or just make separator bars always start from the left margin -- this *seems* like it might be simpler, but I haven't looked at it in detail so I can't say for sure.

If I were more familiar with Rust (and had the time) I would have a go at fixing it myself.
